### PR TITLE
Tweaks to gbstats bandit changes

### PR DESCRIPTION
**Pull request by lukesonnet**

### CODE REVIEW

**Review by lukebrawleysmith — COMMENTED** [review]
added some nits, feel free to merge with or with out them

File: packages/stats/gbstats/models/tests.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def __init__(
         if (
             isinstance(self.stat_b, RegressionAdjustedStatistic)
             and isinstance(self.stat_a, RegressionAdjustedStatistic)
-            and self.stat_a.theta == 0
+            and (self.stat_a.theta is None or self.stat_b.theta is None)
```
[the remark this answers is not in the document]
Comment (lukesonnet) [reply packages/stats/gbstats/models/tests.py:53]: less explicit so i'm going to go with explicit.

File: packages/stats/gbstats/models/statistics.py [hunk 2]
```diff
@@ -121,38 +121,40 @@ class RegressionAdjustedStatistic(Statistic):
...
-    def unadjusted_variances(self):
+    def unadjusted_variances(self) -> float:
         return self.post_statistic.variance
 
     @property
-    def variance(self):
+    def variance(self) -> float:
         if self.n <= 1:
             return 0
+        theta = 0 if self.theta is None else self.theta
```
[the remark this answers is not in the document]
Comment (lukesonnet) [reply packages/stats/gbstats/models/statistics.py:149]: this i like more

File: packages/stats/gbstats/gbstats.py [hunk 3]
```diff
@@ -511,11 +509,13 @@ def process_analysis(
     # Limit to the top X dimensions with the most users
     # not possible to just re-sum for quantile metrics,
     # so we throw away "other" dimension
+    drop_other = (metric.statistic_type not in ["quantile_event", "quantile_unit"]) or (
+        metric.keep_theta and metric.statistic_type == "mean_ra"
+    )
```
[the remark this answers is not in the document]
Comment (lukesonnet) [reply packages/stats/gbstats/gbstats.py:514]: yes ty

File: packages/stats/gbstats/gbstats.py [hunk 4]
```diff
@@ -458,7 +454,9 @@ def variation_statistic_from_metric_row(
         post_pre_sum_of_products = row[f"{prefix}_main_covariate_sum_product"]
         n = row[f"{prefix}_users"]
         # Theta will be overriden with correct value later for A/B tests, needs to be passed in for bandits
-        theta = row[f"{prefix}_theta"] if f"{prefix}_theta" in row.index else 0
+        theta = None
+        if metric.keep_theta:
```
[the remark this answers is not in the document]
Comment (lukesonnet) [reply packages/stats/gbstats/gbstats.py:458]: it won't work due to how get_metric_df sets it to 0, so let's not worry about this rn.

**Review by lukebrawleysmith — APPROVED** [review]

**Pull request merged**